test/conformance/configuration_test.go

```diff
@@ -0,0 +1,139 @@
+	if err != nil {
+		t.Fatalf("Configuration %s was not updated with the new revision: %v", names.Config, err)
+	}
+
+	cfg = reloadConfiguration(names.Config, clients, t)
+
+	logger.Infof("Updating labels of Configuration %s", names.Config)
+	cfg.Labels =  map[string]string{
+		"labelX": "abc",
+		"labelY": "def",
+	}
+	cfg, err = clients.ServingClient.Configs.Update(cfg)
+	if err != nil {
+		t.Fatalf("Failed to update labels for Configuration %s: %v", names.Config, err)
+	}
+
+	if err = waitForConfigurationLabelsUpdate(clients, names, cfg.Labels); err != nil {
+		t.Fatalf("The labels for Configuration %s were not updated: %v", names.Config, err)
+	}
+
+	cfg = reloadConfiguration(names.Config, clients, t)
+	expected := names.Revision
+	actual := cfg.Status.LatestCreatedRevisionName
+	if expected != actual {
+		t.Errorf("Did not expect a new Revision after updating labels for Configuration %s - expected Revision: %s, actual Revision: %s",
+			names.Config, expected, actual)
+	}
+
+	logger.Infof("Updating annotations of Configuration %s", names.Config)
+	cfg.Annotations = map[string]string{
+		"annotationA": "123",
+		"annotationB": "456",
+	}
+	cfg, err = clients.ServingClient.Configs.Update(cfg)
+	if err != nil {
+		t.Fatalf("Failed to update annotations for Configuration %s: %v", names.Config, err)
+	}
+
+	if err = waitForConfigurationAnnotationsUpdate(clients, names, cfg.Annotations); err != nil {
+		t.Fatalf("The annotations for Configuration %s were not updated: %v", names.Config, err)
+	}
+
+	cfg = reloadConfiguration(names.Config, clients, t)
+	actual = cfg.Status.LatestCreatedRevisionName
+	if expected != actual {
+		t.Errorf("Did not expect a new Revision after updating annotations for Configuration %s - expected Revision: %s, actual Revision: %s",
+			names.Config, expected, actual)
+	}
+}
+
+func reloadConfiguration(name string, clients *test.Clients, t *testing.T) *v1alpha1.Configuration {
+	cfg, err := clients.ServingClient.Configs.Get(name, v1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get configuration %s: %v", name, err)
+	}
+	return cfg
+}
+
+func waitForConfigurationLatestCreatedRevision(clients *test.Clients, names test.ResourceNames) (string, error) {
+	var revisionName string
+	err := test.WaitForConfigurationState(clients.ServingClient, names.Config, func(c *v1alpha1.Configuration) (bool, error) {
+		if c.Status.LatestCreatedRevisionName != names.Revision {
+			revisionName = c.Status.LatestCreatedRevisionName
+			return true, nil
+		}
+		return false, nil
+	}, "ConfigurationUpdatedWithRevision")
+	return revisionName, err
+}
+
+func waitForConfigurationLabelsUpdate(clients *test.Clients, names test.ResourceNames, labels map[string]string) error {
+	return test.WaitForConfigurationState(clients.ServingClient, names.Config, func(c *v1alpha1.Configuration) (bool, error) {
+		if reflect.DeepEqual(c.Labels, labels) {
+			return true, nil
+		}
+		return false, nil
```

Comment:
Is this the same as:

```go
return reflect.DeepEqual(c.Labels, labels), nil
```

? Also applies below.

Reply:
It is the same. I will make the changes. Thanks.